tests/bgp/test_bgp_policy.py

```diff
@@ -0,0 +1,554 @@
+    gather_info['duthost'].shell("sudo config interface ip add Loopback13 1.1.3.1/24")
+    gather_info['duthost'].shell("sudo config interface ip add Loopback13 1:1:1:3::/64")
+    gather_info['duthost'].shell("sudo config loopback add Loopback14")
+    gather_info['duthost'].shell("sudo config interface ip add Loopback14 1.1.4.1/24")
+    gather_info['duthost'].shell("sudo config interface ip add Loopback14 1:1:1:4::/64")
+    gather_info['duthost'].shell("sudo config loopback add Loopback15")
+    gather_info['duthost'].shell("sudo config interface ip add Loopback15 1.1.5.1/24")
+    gather_info['duthost'].shell("sudo config interface ip add Loopback15 1:1:1:5::/64")
+
+    # create initial route maps to be used for tests
+    cmd = 'vtysh -c "config" -c "route-map POLICY_TEST_1 permit 10" -c "set community 65001:1" -c "set as-path prepend\
+        11111" -c "route-map POLICY_TEST_2 permit 10" -c "set community 65002:2" -c "set as-path prepend 22222"'
+    gather_info['duthost'].shell(cmd, module_ignore_errors=True)
+
+    cmd = 'vtysh{} -c "config" -c "router bgp {}" -c "address-family ipv4 unicast" \
+          -c "network 1.1.1.0/24 route-map POLICY_TEST_1" -c "network 1.1.2.0/24 route-map POLICY_TEST_1" -c \
+          "network 1.1.3.0/24 route-map POLICY_TEST_2" -c "network 1.1.4.0/24 route-map POLICY_TEST_2" \
+          -c "network 1.1.5.0/24 route-map POLICY_TEST_2" -c "address-family ipv6 unicast" -c \
+          "network 1:1:1:1::/64 route-map POLICY_TEST_1" -c "network 1:1:1:2::/64 route-map POLICY_TEST_1"\
+          -c "network 1:1:1:3::/64 route-map POLICY_TEST_2" -c "network 1:1:1:4::/64 route-map POLICY_TEST_2" -c \
+          "network 1:1:1:5::/64 route-map POLICY_TEST_2"'.format(
+              gather_info['cli_options'], gather_info['dut_asn'])
+    logger.debug(gather_info['duthost'].shell(cmd, module_ignore_errors=True))
+    logger.debug(gather_info['duthost'].shell("show ip bgp summary", module_ignore_errors=True)['stdout'])
+
+    # Verify loopbacks are being advertised to neighbor
+    output = gather_info['duthost'].shell('show ip bgp neighbors {} advertised-routes | grep "*> 1.1."'.format(
+        gather_info['neigh_ip_v4']), module_ignore_errors=True)['stdout'].split("\n")
+    out_num = len(output)
+    assert 5 == out_num
+    logger.debug("out split: {}\nbase: {} num: {}".format(output, num_v4_routes, out_num))
+    output = gather_info['duthost'].shell('show ipv6 bgp neighbors {} advertised-routes | grep "*> 1:1:"'.format(
+        gather_info['neigh_ip_v6']), module_ignore_errors=True)['stdout'].split("\n")
+    out_num = len(output)
+    assert 5 == out_num
+    logger.debug("out split: {}\nbase: {} num: {}".format(output, num_v6_routes, out_num))
+    output = gather_info['duthost'].shell('show ip bgp neighbors {} advertised-routes | grep -c "*>"'.format(
+        gather_info['neigh_ip_v4']), module_ignore_errors=True)['stdout']
+    assert 5 + int(num_v4_routes) == int(output)
```

Comment:
This part (lines 155~162) seems to be doing something similar to the check_baseline function defined above. Why don't you just invoke check_baseline() function here?

Reply:
These lines are explicitly looking for the 1.1 or 1:1: routes, whereas baseline looks at all routes.

Reply:
Sorry. I am referring to lines 159~165. They are checking the count of all routes.

Reply:
Correct, but the baseline routes will not appear in this list after the configuration changes.  So it needs to check for only the 1.1 or 1:1: routes, not all of them as is done in the function.

Reply:
If not all the routes are displayed, why are you asserting "5 + int(num_v4_routes) == int(output)"? num_v4_routes is the number of total IPv4 routes, isn't it?

Reply:
This must have been missed in a refactor.  You are correct that this block is checking what the function does.  This has been rewritten to uses the function.